app/controllers/miq_ae_class_controller.rb

```diff
@@ -197,6 +200,20 @@ def get_node_info(node)
     x_history_add_item(:id => x_node, :text => @right_cell_text)
   end
 
+  def add_all_domains_version_message(domains)
+    @version_messages = []
+    domains.collect { |dom| [dom.name, dom.version, dom.available_version] }.each do |name, current, available|
+      @version_messages.push(_("%s domain: Current version - %s  Available version - %s") % [name, current, available]) if !current.nil? && !available.nil? && current != available
+    end
+  end
+
+  def add_domain_version_message
```

Comment:
@h-kataria 
These methods could be combined into something like this

def domain_version_message(domain)
  version = domain.version
  available_version = domain.available_version
  return if version.nil? || available_version.nil?
  _("Currently running %s version, you have %s version available.") % [version, available_version] if version != available_version
end

def add_all_domains_version_message(domains)
  @version_messages = []
   domains.each do |dom|
     domain_version_message(dom).tap { |msg| @version_messages.push << msg if msg }
  end
end